pkg/catalog/catalog.go

```diff
@@ -1894,9 +1938,36 @@ func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID string,
 	}()
 
 	uw := NewUncommittedWriter(fd)
+	pw, err := writer.NewParquetWriterFromWriter(uw, new(UncommittedParquetObject), gcParquetParallelNum) // TODO: Play with np count
+	if err != nil {
+		return nil, err
+	}
+	pw.CompressionType = parquet.CompressionCodec_GZIP
 
 	// Write parquet to local storage
-	newMark, hasData, err := c.writeUncommittedLocal(ctx, repository, uw, mark, runID)
+	hasData := false
+	var newMark *GCUncommittedMark
+	if mark != nil && mark.BranchID == "" {
+		newMark, hasData, err = c.writeTokensLocal(ctx, repository, uw, pw, mark, runID)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		newMark, hasData, err = c.writeUncommittedLocal(ctx, repository, uw, pw, mark, runID)
+		if err != nil {
+			return nil, err
+		}
+
+		if newMark == nil {
+			// TODO(in progress) - fix hasData
```

Comment:
You can pass hadData as reference to the writeTokensLocal and "modify" it to true if we've written token data

Reply:
done